pkg/cmd/dev/dev.go

```diff
@@ -124,3 +169,69 @@ func findModelFile(absPath string) (string, error) {
 	output.Debugf("Found model path in directory %s at %s", absPath, modelPath)
 	return modelPath, nil
 }
+
+// extractModelKitToCache extracts a ModelKit reference to a cache directory
+// using the unpack library with model filter
+func extractModelKitToCache(ctx context.Context, options *DevStartOptions) error {
+	output.Infof("Extracting ModelKit %s to cache directory...", options.modelRef.String())
+
+	// Use consistent cache directory for extraction
+	extractDir := filepath.Join(options.configHome, "dev-models", "current")
+	if err := os.MkdirAll(extractDir, 0755); err != nil {
+		return fmt.Errorf("failed to create cache directory: %w", err)
+	}
+	options.contextDir = extractDir
+
+	// Extract the ModelKit using the library directly
+	libOpts := &unpack.UnpackOptions{
+		ModelRef:       options.modelRef,
+		UnpackDir:      extractDir,
+		ConfigHome:     options.configHome,
+		Overwrite:      true, // Safe for extraction directory
+		NetworkOptions: options.NetworkOptions,
+	}
+
+	// Add model filter
+	modelFilter, err := unpack.ParseFilter("model,kitfile")
+	if err != nil {
+		return fmt.Errorf("failed to create model filter: %w", err)
+	}
+	libOpts.FilterConfs = []unpack.FilterConf{*modelFilter}
+
+	// Change working directory to cache directory unpack logic is relative to CWD
+	originalWd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current working directory: %w", err)
+	}
+	if err := os.Chdir(extractDir); err != nil {
+		return fmt.Errorf("failed to change to cache directory %s: %w", extractDir, err)
+	}
+	// Restore original working directory when done
+	defer func() {
+		if err := os.Chdir(originalWd); err != nil {
+			output.Debugf("Failed to restore working directory: %v", err)
+		}
+	}()
```

Comment:
This looks like something that should be managed inside the `unpack` library function, given that it takes an `UnpackDir` parameter.